Add log out button

anirudh723 opened the pull request:
Closes issue #67 in [LLB frontend](https://github.com/Code-4-Community/lucys-love-bus-frontend/issues/67).

This PR adds a log out button to the dropdown of the navigation bar. I added a "Log Out" menu item, and upon click, it removes the state in the localStorage, and routes the user to the home page. 

Screenshots:

Log out button:
![image](https://user-images.githubusercontent.com/46979800/111922889-e81c3680-8a72-11eb-90f8-569644fd2210.png)

Before logging in (localStorage is empty):
![image](https://user-images.githubusercontent.com/46979800/111922905-f9fdd980-8a72-11eb-935f-89b70b1a54d3.png)

After logging in (localStorage is updated with state):
![image](https://user-images.githubusercontent.com/46979800/111922915-0aae4f80-8a73-11eb-943d-86ac2ca77324.png)

After clicking the newly added log out button, back at the home page (localStorage is cleared out):
![image](https://user-images.githubusercontent.com/46979800/111922969-34677680-8a73-11eb-991f-c4c81fa503e4.png)

cc: @varunthakkar1

anirudh723 committed:
add logout button + plus functionality

anirudh723 committed:
updated comment

jackblanc commented on src/components/navbar/index.tsx:
```diff
@@ -94,6 +96,19 @@ const NavBar: React.FC<NavBarProps> = ({ tokens }) => {
+          if (asyncRequestIsComplete(tokens)) {
+            AuthClient.logout(tokens.result.refreshToken).then(() => {
+              localStorage.removeItem(LOCALSTORAGE_STATE_KEY);
+            });
```
This should be done by `dispatch(logoutUser())`, we have a thunk which makes this request and clears the store.

jackblanc reviewed (changes requested):
You've got the right idea, but a lot of this logic already lives in a thunk, logoutUser. We should continue to use that thunk so that logout behavior is the same in any place we call it.

rymaju commented:
https://github.com/Code-4-Community/frontend-scaffold/pull/39

Is relevant here, you can now use logout and it will clear tokens for you!

rymaju commented on src/components/navbar/index.tsx:
```diff
@@ -94,6 +96,19 @@ const NavBar: React.FC<NavBarProps> = ({ tokens }) => {
+              localStorage.removeItem(LOCALSTORAGE_STATE_KEY);
+            });
+            history.push(Routes.HOME);
+            history.go(0); // force refresh if user is already on home page
```
force refresh here is actually pretty big brain, I kind of like it 😄 

I think we should still use the thunk from master but this is super creative

anirudh723 committed:
Merge branch 'master' into AS.add-logout-button

anirudh723 committed:
dispatched to our logout thunk

rymaju commented on src/components/navbar/index.tsx:
```diff
@@ -94,6 +97,17 @@ const NavBar: React.FC<NavBarProps> = ({ tokens }) => {
+          if (asyncRequestIsComplete(tokens)) {
+            dispatch(logout());
+            history.push(Routes.HOME);
+            history.go(0); // force refresh if user is already on home page
```
⛏️ We dont _need_ to force refresh anymore, I dont really mind if we do though 🤷

anirudh723 committed:
remove force refresh

Removed the force refresh, even though it looks kind of odd with a reload. When sitting on the home page already, and logging out, there's just a split second of button change in the navbar, so it's not a very visually pleasing log out. The reload made it clear that we logged out. Easy to put back in if we wish though

jackblanc reviewed (approved):
Nice work. For bonus points (we can pair on this) we should add a Jest/Enzyme test for the button click. That's not something we've done before, but would be beneficial to start.
